src/or/proto_socks.c

```diff
@@ -378,12 +476,26 @@ handle_socks_message(const uint8_t *raw_data, size_t datalen, socks_request_t *r
       res = 0;
       goto end;
     }
+    /* RFC1929 SOCKS5 username/password subnegotiation. */
+    if ((!req->got_auth && raw_data[0] == 1) ||
```

Comment:
Should be a constant?

Reply:
805484e2be5ded2736fd3ace42aa804da4f3e638